File: leetcode/java/1603. Design Parking System.java
class ParkingSystem {

    private final Map<Integer, Integer> parking = new HashMap<>();
    
    public ParkingSystem(int big, int medium, int small) {
        parking.put(1, big);
        parking.put(2, medium);
        parking.put(3, small);
    }
    
    public boolean addCar(int carType) {
        final int size = parking.get(carType);
        
        if(size <= 0) {
            return false;
        }
        
        parking.put(carType, size - 1);
        return true;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */

File: leetcode/kotlin/1603. Design Parking System.kt
class ParkingSystem(big: Int, medium: Int, small: Int) {

    private var store = mutableMapOf<Int, Int>(1 to big, 2 to medium, 3 to small)
    
    fun addCar(carType: Int): Boolean {
        val size = store.get(carType)
        return if(size == 0) false else {
            store.put(carType, size!!.minus(1))
            return true
        }
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */
